aws-cpp-sdk-transfer/source/transfer/TransferManager.cpp

```diff
@@ -65,6 +65,59 @@ namespace Aws
             }
         }
 
+        Transfer::TransferStatus TransferManager::WaitUntilAllFinished(int64_t timeoutMs)
+        {
+            const auto waitStarted = std::chrono::steady_clock::now();
+            Transfer::TransferStatus status = Transfer::TransferStatus::IN_PROGRESS;
+            {
+                std::unique_lock<std::mutex> lock(m_tasksMutex);
```

Comment:
nit: this looks almost identical to the body of the while if you switch wait_for order